plugins/inputs/windows_event_log/wineventlog/utils.go

```diff
@@ -112,8 +105,31 @@ func UTF16ToUTF8Bytes(in []byte, length uint32) ([]byte, error) {
 	return decoded, err
 }
 
+func UTF16ToUTF8BytesForWindowsEventBuffer(in []byte, length uint32) ([]byte, error) {
+	// Since Windows server 2022, the returned value of used buffer represents for double bytes char count,
+	// which is half of the actual buffer used by byte(what older Windows OS returns), checking if the length
+	//land on the end of used buffer, if no, double it.
+	if NumbersOfBytesPerCharacter == UnknownBytesPerCharacter {
```

Comment:
I haven't looked at the code much but my concern here is concurrency - though I think maybe it's not a super big issue since it would be on the same machine so the outcome would (theoretically) be the same. Is there any potential for a race condition when trying to evaluate `NumberOfBytesPerCharacter` when calling this function? 

When I think about it, I don't think there's really much harm, but figured I'd bring it up just in case.